### PR TITLE
Add Sprint support (G920P)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: rootdir/Android.mk [hunk 1]
```diff
@@ -1,10 +1,26 @@
 LOCAL_PATH := $(call my-dir)
 
 include $(CLEAR_VARS)
-LOCAL_MODULE		:= fstab.samsungexynos7420
+LOCAL_MODULE            := fstab.goldfish
```
Comment (bmc08gt) [inline rootdir/Android.mk:4]: not needed.

File: rootdir/Android.mk [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
+include $(CLEAR_VARS)
+LOCAL_MODULE            := fstab.samsungexynos7420
+LOCAL_MODULE_TAGS       := optional eng
+LOCAL_MODULE_CLASS      := ETC
+LOCAL_SRC_FILES         := etc/fstab.samsungexynos7420
+LOCAL_MODULE_PATH       := $(TARGET_ROOT_OUT)
+include $(BUILD_PREBUILT)
+
+include $(CLEAR_VARS)
+LOCAL_MODULE		:= fstab.samsungexynos7420.fwup
```
Comment (bmc08gt) [inline rootdir/Android.mk:20]: not needed. Used by samsung for fw updates as failsafe

File: rootdir/Android.mk [hunk 3]
```diff
@@ -25,6 +41,22 @@ LOCAL_MODULE_PATH       := $(TARGET_ROOT_OUT)
 include $(BUILD_PREBUILT)
 
 include $(CLEAR_VARS)
+LOCAL_MODULE            := init.container.rc
```
Comment (bmc08gt) [inline rootdir/Android.mk:44]: Not needed. KNOX mountpoints

File: rootdir/Android.mk [hunk 4]
```diff
@@ -25,6 +41,22 @@ LOCAL_MODULE_PATH       := $(TARGET_ROOT_OUT)
...
+LOCAL_MODULE_TAGS       := optional eng
+LOCAL_MODULE_CLASS      := ETC
+LOCAL_SRC_FILES         := etc/init.container.rc
+LOCAL_MODULE_PATH       := $(TARGET_ROOT_OUT)
+include $(BUILD_PREBUILT)
+
+include $(CLEAR_VARS)
```
Comment (bmc08gt) [inline rootdir/Android.mk:51]: Not needed.

File: rootdir/Android.mk [hunk 5]
```diff
@@ -57,6 +113,22 @@ LOCAL_MODULE_PATH	:= $(TARGET_ROOT_OUT)
...
 
 include $(CLEAR_VARS)
+LOCAL_MODULE            := init.recovery.samsungexynos7420.rc
+LOCAL_MODULE_TAGS       := optional eng
+LOCAL_MODULE_CLASS      := ETC
+LOCAL_SRC_FILES         := etc/init.recovery.samsungexynos7420.rc
+LOCAL_MODULE_PATH       := $(TARGET_ROOT_OUT)
+include $(BUILD_PREBUILT)
+
+include $(CLEAR_VARS)
```
Comment (bmc08gt) [inline rootdir/Android.mk:123]: Not needed

File: rootdir/Android.mk (hunk 1 above)
Comment (ghost) [inline rootdir/Android.mk:4]: Even for emulation?

File: rootdir/Android.mk (hunk 2 above)
Comment (ghost) [inline rootdir/Android.mk:20]: Will remove.

File: rootdir/Android.mk (hunk 3 above)
Comment (ghost) [inline rootdir/Android.mk:44]: Ah.  Will remove.

File: rootdir/Android.mk (hunk 4 above)
Comment (ghost) [inline rootdir/Android.mk:51]: Will remove.

File: rootdir/Android.mk (hunk 5 above)
Comment (ghost) [inline rootdir/Android.mk:123]: Can you explain why this one isn't needed?

**Comment by bmc08gt** [comment]
Some of this (chmod +x, nuking uncommon files) was done recently. Rebase and recreate pull request, after additionally fixing above.

File: rootdir/Android.mk (hunk 1 above)
Comment (bmc08gt) [inline rootdir/Android.mk:4]: It's compiled and included always. No reason to have it be a prebuilt.

**Comment by ghost** [comment]
The SPR version appears a different size than the others unless I'm miscalculating the block sizes:

--- Entry #21 ---
Binary Type: 0 (AP)
Device Type: 8 (Unknown)
Identifier: 18
Attributes: 5 (Read/Write)
Update Attributes: 5 (FOTA)
Partition Block Size/Offset: 57600
Partition Block Count: 957440
File Offset (Obsolete): 0
File Size (Obsolete): 0
Partition Name: SYSTEM
Flash Filename: system.img
FOTA Filename: 

which comes out to be 3921674240 bytes (957440*4096).  Is there an overhead I'm missing?

**Comment by bmc08gt** [comment]
System partition needs to be the max size of all variants or else we will have issues with size. (ie TMO is bigger than you are giving it space for).

**Comment by ghost** [comment]
Ah, ok.  I'll do as requested and start from scratch, comment the areas I see different with SPR and recommit.

**Pull request closed without merging**